src/bundle/Controller/LocationController.php

```diff
@@ -43,17 +44,20 @@ public function __construct(
     }
 
     public function contentViewTabAction(
-        ContentView $view
+        ContentView $view,
+        LocationParameterSupplier $locationParameterSupplier
     ) {
         $versionInfo = $view->getContent()->getVersionInfo();
         $contentInfo = $versionInfo->getContentInfo();
 
         if ($contentInfo->published) {
-            $locations = $this->uiLocationService->loadLocations($contentInfo);
-            $actionsForm = $this->formFactory->createLocationsActionForm($locations);
+            $locationParameterSupplier->supply($view);
+
+            $actionsForm = $this->formFactory->createLocationsActionForm(
+                $view->hasParameter('locations') ? $view->getParameter('locations') : []
```

Comment:
I understand why you would reuse that value, but it really feels like a sever break of responsibilities.